### PR TITLE
[luci] Validate for number of inputs/outputs

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce validate method to check number of inputs and single
output.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #6053, from draft #6054

File: compiler/luci/import/include/luci/Import/GraphBuilder.h [hunk 1]
```diff
@@ -33,6 +33,12 @@ class GraphBuilder : public GraphBuilderBase
 public:
   virtual ~GraphBuilder() = default;
 
+  // common validate to check number of inputs and single output
+  bool validate(const ValidateArgs &args, size_t i) const
```
Comment (llFreetimell) [inline compiler/luci/import/include/luci/Import/GraphBuilder.h:37]: How about `validate_input_cnt` or `validate_common_inout` or `is_single_output_with` or something else?
`validate` seems generous.

Comment (seanshpark) [reply compiler/luci/import/include/luci/Import/GraphBuilder.h:37]: Yes, it seems general... 
Behind this is that I like short names and `size_t i` argument will distinguish existing method...

File: compiler/luci/import/include/luci/Import/GraphBuilder.h [hunk 2]
```diff
@@ -33,6 +33,12 @@ class GraphBuilder : public GraphBuilderBase
 public:
   virtual ~GraphBuilder() = default;
 
+  // common validate to check number of inputs and single output
```
Comment (llFreetimell) [inline compiler/luci/import/include/luci/Import/GraphBuilder.h:36]: Maybe :D ?
```suggestion
  // common validation to check number of inputs and single output
```

File: compiler/luci/import/include/luci/Import/GraphBuilder.h (hunk 1 above)
Comment (llFreetimell) [reply compiler/luci/import/include/luci/Import/GraphBuilder.h:37]: I see it :)
Then `bool validate(const ValidateArgs &args, size_t input_cnt) const` seems okay?
I like short names too but if both function name and parameter name iare too short, sometimes it is confused :(
If you don't like this too, let's go as it is :)

File: compiler/luci/import/include/luci/Import/GraphBuilder.h (hunk 2 above)
Comment (seanshpark) [reply compiler/luci/import/include/luci/Import/GraphBuilder.h:36]: Ah, `validate` is the method `validate`... I'll revise this...

File: compiler/luci/import/include/luci/Import/GraphBuilder.h (hunk 1 above)
Comment (seanshpark) [reply compiler/luci/import/include/luci/Import/GraphBuilder.h:37]: `input_cnt` seems OK :)

**Review by llFreetimell — APPROVED** [review]
LGTM

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**